scheduler/strategy/weighted_node.go

```diff
@@ -71,3 +73,58 @@ func weighNodes(config *cluster.ContainerConfig, nodes []*node.Node) (weightedNo
 
 	return weightedNodes, nil
 }
+
+func ignoreInactiveWeighNodes(config *cluster.ContainerConfig, nodes []*node.Node) (weightedNodeList, error) {
+	weightedNodes := weightedNodeList{}
+	log.Debug("In  ignoreInactive weigh nodes...")
+	for _, node := range nodes {
+		nodeMemory := node.TotalMemory
+		nodeCpus := node.TotalCpus
+
+		// Skip nodes that are smaller than the requested resources.
+		if nodeMemory < int64(config.Memory) || nodeCpus < config.CpuShares {
+			continue
+		}
+
+		var (
+			cpuScore         int64 = 100
+			memoryScore      int64 = 100
+			notRunningSahres int64
+			notRunningMemory int64
+		)
+
+		for _, container := range node.Containers {
+			if container.Status != "running" {
+
+				//				container.State
+
+				notRunningSahres += container.Config.CpuShares
+				notRunningMemory += container.Config.Memory
+
+				//				notRunningSahres += container.Info.Config.CpuShares
+				//				notRunningMemory += container.Info.Config.Memory
+			}
+
+		}
+
+		log.Debug("Deducing cpu shres", notRunningSahres)
+		log.Debug("Deducing memory shres", notRunningMemory)
+
+		if config.CpuShares > 0 {
+			cpuScore = (node.UsedCpus + config.CpuShares - notRunningSahres) * 100 / nodeCpus
```

Comment:
This causes over-commit problem. That means the stopped containers cannot start (at least with the resource reservation). We need some consensus on this.